webapp/src/main/java/org/apache/atlas/web/rest/AlternateREST.java

```diff
@@ -0,0 +1,116 @@
+        if (!ARGO_SERVICE_USER_NAME.equals(RequestContext.getCurrentUser())) {
+            throw new AtlasBaseException(AtlasErrorCode.UNAUTHORIZED_ACCESS, RequestContext.getCurrentUser(), "Policy linking");
+        }
+
+        // Set request context parameters
+        RequestContext.get().setAlternatePath(true);
+        RequestContext.get().setIncludeClassifications(false);
+        RequestContext.get().setIncludeMeanings(false);
+
+        AtlasPerfTracer perf = null;
+        try {
+            // Start performance tracing if enabled
+            if (AtlasPerfTracer.isPerfTraceEnabled(PERF_LOG)) {
+                perf = AtlasPerfTracer.getPerfTracer(PERF_LOG, "AlternateREST.linkBusinessPolicy(" + policyId + ")");
+            }
+
+            // Link the business policy to the specified entities
+            entitiesStore.linkBusinessPolicy(policyId, request.getLinkGuids());
+        } catch (AtlasBaseException abe) {
+            LOG.error("Error in policy linking: ", abe);
+            throw new AtlasBaseException(AtlasErrorCode.INTERNAL_ERROR, "During Policy linking");
+        } finally {
+            // Log performance metrics
+            AtlasPerfTracer.log(perf);
+        }
+    }
+
+    /**
+     * Unlinks a business policy from entities.
+     *
+     * @param policyId the ID of the policy to be unlinked
+     * @param request  the request containing the GUIDs of the assets to unlink the policy from
+     * @throws AtlasBaseException if there is an error during the unlinking process
+     */
+    @POST
+    @Path("/{policyId}/unlink-business-policy")
+    @Timed
+    public void unlinkBusinessPolicy(@PathParam("policyId") final String policyId, final LinkBusinessPolicyRequest request) throws AtlasBaseException {
+        // Ensure the current user is authorized to unlink policies
+        if (!ARGO_SERVICE_USER_NAME.equals(RequestContext.getCurrentUser())) {
+            throw new AtlasBaseException(AtlasErrorCode.UNAUTHORIZED_ACCESS, RequestContext.getCurrentUser(), "Policy unlinking");
+        }
+
+        // Set request context parameters
+        RequestContext.get().setAlternatePath(true);
+        RequestContext.get().setIncludeClassifications(false);
+        RequestContext.get().setIncludeMeanings(false);
+
+        AtlasPerfTracer perf = null;
+        try {
+            // Start performance tracing if enabled
+            if (AtlasPerfTracer.isPerfTraceEnabled(PERF_LOG)) {
+                perf = AtlasPerfTracer.getPerfTracer(PERF_LOG, "AlternateREST.unlinkBusinessPolicy(" + policyId + ")");
+            }
+
+            // Unlink the business policy from the specified entities
+            entitiesStore.unlinkBusinessPolicy(policyId, request.getUnlinkGuids());
+        } catch (AtlasBaseException abe) {
+            LOG.error("Error in policy unlinking: ", abe);
+            throw new AtlasBaseException(AtlasErrorCode.INTERNAL_ERROR, "During Policy unlinking");
```

Comment:
we are logging it so that we don't lose track of it, always the better practice to throw a custom exception so that the consumer is not aware of the internals of the system, if we agree to this we will keep it as is otherwise I will make the change

Reply:
https://github.com/atlanhq/atlas-metastore/pull/3349#discussion_r1689163825